### DISCUSSION TITLE
tRPC for both Electron IPC and HTTP API routes at the same time

### DISCUSSION THREAD

**Discussion by harrismcc** [trpc/trpc#4675] [Q&A]
I'm working in a monorepo right now with a nextjs web app/ backend, and an electron desktop app that also uses the nextjs backend (all running trpc). 

I want to use tRPC for Electron IPC (to talk between the host machine and the react app) and I'm using an awesome library [electron-trpc](https://github.com/jsonnull/electron-trpc) by @jsonnull to do it. This works super well and I happy with it.

The issue is, I want to have a separate tRPC client for both my regular API calls to the backend, as well as my IPC calls. I found [this test](https://github.com/trpc/trpc/blob/176c0dbeb420cb8bd946bd2338eca0b0e41c85e8/packages/tests/server/react/multiple-trpc-providers.test.tsx) in this repo that seems to imply that you can have multiple providers, but I can't seem to get it working.

Has anyone tried this use case before? Is there a better way that I'm not aware of? Thanks!

**Comment by harrismcc** [comment]
To clarify here, I'm asking more about having more than one provider that I am about the electron IPC stuff specifically.

**Reply by jsonnull** [reply]
Hey, I'd love to help. I think what you're trying to do should work. Do you think you could share what you've been trying so I can help diagnose what might be going wrong?
